samplePDF/CMakeLists.txt

```diff
@@ -2,27 +2,33 @@ set(SamplePDF_implementation_files
 	samplePDFBase.cpp 
 	samplePDFFDBase.cpp 
 	interfacePDFEbE.cpp 
-	Structs.cpp
-        probGpu.cu )
-        
+	Structs.cpp)
 
 set(SamplePDF_header_files 
 	samplePDFBase.h
 	samplePDFFDBase.h
 	interfacePDFEbE.h
 	Structs.h )
 
-add_library(SamplePDF SHARED ${SamplePDF_implementation_files})
+if(NOT ${CPU_ONLY})
+  list(APPEND SamplePDF_implementation_files probGpu.cu)
+endif()
```

Comment:
This is bit messy but you right now you compile probGpu even if using CUDAprob. I know this should be fixed with Dan't oscialtor but all this stuff should be moved from SamplePDF

Reply:
Are you happy if we leave this for now?

Reply:
It's ugly but doesn't break any funcionality so yes, let's leave it